### DISCUSSION TITLE
AI model - self driving car

### DISCUSSION THREAD

**Discussion by BeirlaenSander** [carla-simulator/carla#7431] [Q&A]
I'm doing a bachelor about Carla and selfdriving cars. Maybe a bit of an ambitious project since it requires a lot of hardware but that asside i've got a couple of models and was wondering which one would be best. Perhaps even a different one...

First of DQN, i've seen Sentdex using this but so far i've had no luck.
So i switched to using these models:
- Deep determentistic Policy gradiënt
- Soft Actor Critic
- Proximal Policy Optimitzation

I'm trying to use an RGB camera so it can drive between the lines and don't collide with other objects. Does anyone have an idea on which reinforcement model i need to focus?

Thx for the help!

**Comment by joschi27** [comment]
Hey man, we're doing the same thing basically, but just for a racecar :)

For us, this is what worked:
- proximal policy optimization (a basic implementation from github) 
- training a separate YOLO model for perception tasks, as the perception task is very very hard to learn when having the model learn everything else aswell

We were able to train the model on a single (although beefy) computer. To have a model that drives well, it trains for about 3 days.

Hope that helps somewhat, good luck

**Reply by BeirlaenSander** [reply]
Thanks for the reply, this helps me a lot! Is there any chance that you have the specs of the graphics card?
